### PR TITLE
(PC-17107) feat(search): use fullscreen modal price filter

### PR DESCRIPTION
**Pull request by cedricls1**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-17107

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|Capture d’écran 2022-09-16 à 17 15 06|Capture d’écran 2022-09-16 à 17 26 03|Capture d’écran 2022-09-16 à 17 14 12|Capture d’écran 2022-09-16 à 17 14 21|                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/search/components/SearchResults.tsx [hunk 1]
```diff
@@ -120,8 +126,8 @@ export const SearchResults: React.FC = () => {
   }, [showCategoriesModal])
 
   const redirectToPriceFilterPage = useCallback(() => {
-    navigate('SearchPrice')
-  }, [navigate])
+    showSearchPriceModal()
```
Comment (kopax-polyconseil) [inline src/features/search/components/SearchResults.tsx:129]: on peut pas virer ça et juste mettre un `showSearchPriceModal()` ? ou le typing n'aimera pas ?

File: src/features/search/pages/Categories.tsx [hunk 2]
```diff
@@ -67,40 +67,24 @@ export const Categories: FunctionComponent<Props> = ({
...
-    ),
-    [onResetPress, onSearchPress]
-  )
-
   return (
     <AppModal
       visible={isVisible}
-      customModalHeader={CustomModalHeaderComponent}
+      customModalHeader={
+        isDesktopViewport ? undefined : (
```
Comment (kopax-polyconseil) [inline src/features/search/pages/Categories.tsx:74]: `isDesktopViewport && <SearchCustomModalHeader titleId={titleId} title={title} onGoBack={hideModal} />`  devrait fonctionné non ?

File: src/features/search/components/SearchResults.tsx (hunk 1 above)
Comment (cedricls1) [reply src/features/search/components/SearchResults.tsx:129]: Bien vu

File: src/features/search/pages/Categories.tsx (hunk 2 above)
Comment (cedricls1) [reply src/features/search/pages/Categories.tsx:74]: ça ne fonctionne pas, il y a l'erreur suivante :

`Impossible d'assigner le type 'false | Element | undefined' au type 'Element | undefined'.
  Impossible d'assigner le type 'boolean' au type 'Element'.`

**Review by kopax-polyconseil — APPROVED** [review]

**Pull request merged**